### PR TITLE
Feature/validate request screen

### PR DESCRIPTION
**Pull request by Thibaud-Babin**
##  Summary
Implement the ValidateRequestScreen UI for request closure and kudos attribution to helpers.

## Why
The core goal of the app was to be able to reward people for their help, and without this the app was well... kind of pointless. This PR aims to implement this core feature!

**Note**: Navigation integration will be addressed in a follow-up PR.

## Addresses issue
- #190 
## Addresses user story
- #199 
## Follows PR
- #223 
---

### Core Implementation
* Complete `ValidateRequestScreen` composable with state-driven architecture
* `ValidateRequestViewModel` for managing validation business logic
* `ValidateRequestCallbacks` data class for grouped callback handling

### UI Features (shown on previews)
* **Loading State**: Progress indicator while fetching request data
* **Ready State**: 
  * Display list of helpers who accepted the request
  * Selectable helper cards with animated selection indicators
  * Real-time kudos summary showing total points to award
  * Empty state handling when no helpers joined
* **Confirming State**: Confirmation dialog showing selected helpers and kudos breakdown
* **Processing State**: Loading indicator during request closure
* **Success State**: Success message with automatic navigation trigger
* **Error State**: Error message with optional retry button

### Helper Selection
* Animated card selection with color transitions
* Profile pictures for each helper
* Kudos badge showing points each helper will receive
* Multi-select support for crediting multiple helpers

### Architecture
* **State hoisting pattern**: Screen accepts state and callbacks as parameters
* **Testable design**: Pure composable function with no direct ViewModel dependency
* **Callback grouping**: `ValidateRequestCallbacks` data class to reduce parameter count (SonarCloud compliant)

---

### Test Infrastructure
* Comprehensive instrumented test suite (`ValidateRequestScreenTest.kt`)
* Unit tests for ViewModel (`ValidateRequestViewModelTest.kt`)
* `FakeUserProfileRepository` for isolated testing without Firebase
* `CallbackTracker` for verifying UI interactions
* `TestDataFactory` for consistent test data generation
* 
### Test Design
* All magic values extracted to named constants
* Modular structure for easy test expansion
* Fake implementations instead of mocks for Android instrumented test compatibility

---

### Design Decisions
* **State/Callbacks pattern over ViewModel injection**: Enables pure UI testing without mocking ViewModels
* **Fake implementations over mocks**: Avoids MockK proxy issues with interfaces in instrumented tests
* **Sealed class for states**: Exhaustive `when` expressions ensure all states are handled
* **Callback grouping**: `ValidateRequestCallbacks` keeps parameter count under SonarCloud's limit of 7

---

##  Previews
12 preview composables included covering all states:
* Loading State
* Ready State (No Selection, With Selection, All Selected, No Helpers)
* Confirming State (With Helpers, No Helpers)
* Processing State
* Success State
* Error State (With Retry, No Retry)
* I didn't give a screen recoring as integration into the project will be done in a follow up PR

image
image
image

--

### Follow-up PR
* Navigation graph integration
* Connection to request detail screen
* Deep linking support

### Kudos System
* Only helpers receive kudos upon request closure
* Creator does not receive bonus kudos (by design)

### AI Usage Disclosure
 **AI tools were used** to assist with writing code, tests, and this PR description.

### SonarCloud
No sonar cloud issue to report (yay)

### CODE REVIEW

File: app/src/androidTest/java/com/android/sample/ui/validate/ValidateRequestScreenTests.kt [hunk 1]
```diff
@@ -0,0 +1,813 @@
...
+
+    // When
+    composeTestRule.onNodeWithText(ValidateRequestConstants.BUTTON_GO_BACK).performClick()
+
+    // Then
+    assertEquals(TestAssertions.SINGLE_INVOCATION, callbackTracker.navigateBackCount)
+  }
+
+  // ==================== EDGE CASE TESTS ====================
+
```
Comment (unusedusername01) [inline app/src/androidTest/java/com/android/sample/ui/validate/ValidateRequestScreenTests.kt]: > [!NOTE] 
> Good coverage of edge cases! However, consider adding tests for:
> - A very high number of helpers (to test performances, I guess a handsome composer such as yourself would have no troubles imagining a scenario where the whole campus wants to be your lunch buddy)
> - Helper with zero kudos :'(
> - Request with very long description (since you would need to scroll a lot to reach the volunteers tab)
> But other than that great tests !

File: app/src/androidTest/java/com/android/sample/utils/FirebaseEmulator.kt [hunk 2]
```diff
@@ -85,6 +86,34 @@ object FirebaseEmulator {
     Log.d("FirebaseEmulator", "Signed in as ${auth.currentUser?.uid}")
   }
 
+  /** Add a test request to Firestore for testing purposes. */
+  suspend fun addTestRequest(
```
Comment (unusedusername01) [inline app/src/androidTest/java/com/android/sample/utils/FirebaseEmulator.kt]: > [!WARNING]
> You use magic value 8.64e7 (einstein notation type shi)
> What are they ? Are they 8.64e7 Apples ? Bananas ? Dollars evaded from taxation on the 2024-2025 fiscal year ? Just kidding you added a comment and you used it once so I won't hold grudge but I promise you you don't want to see the ugly side of me 😠💪 
> Also you call System.currentTimeMillis() 2 separate times; it is better to define it once beforehand and use it the 2 times

> [!NOTE]
> Other than that adding the implementation of this util and its placement is very well thought; We finally have something to avoid redefining the same function 40 times over

File: app/src/main/java/com/android/sample/ui/request_validation/ValidateRequestScreen.kt [hunk 3]
```diff
@@ -0,0 +1,696 @@
...
+ * and award kudos.
+ *
+ * @param requestId The ID of the request to validate
+ * @param viewModel The ViewModel managing the validation state
+ * @param userProfileRepository Repository for loading user profiles (for ProfilePicture)
+ * @param onRequestClosed Callback invoked when request is successfully closed
+ * @param onNavigateBack Callback to navigate back
+ */
+@Composable
+fun ValidateRequestScreen(
```
Comment (unusedusername01) [inline app/src/main/java/com/android/sample/ui/request_validation/ValidateRequestScreen.kt]: > [!NOTE]
> SonarCloud flags this as it has 10 paremeters (greater than 7) but I will accept it for this time like a good boy 
> Maybe in the future we could define an object e.g. `TriggerActions` to merge some of the on... since many of them are reused throughout the code ? But I'm not sure we'll be needing it anytime soon

File: app/src/main/java/com/android/sample/ui/request_validation/ValidateRequestScreen.kt [hunk 4]
```diff
@@ -0,0 +1,696 @@
...
+            style = MaterialTheme.typography.titleLarge,
+            fontWeight = FontWeight.Bold)
+      },
+      navigationIcon = {
+        if (canNavigateBack) {
+          IconButton(
+              onClick = onNavigateBack,
+              modifier = Modifier.testTag(ValidateRequestConstants.TAG_BACK_BUTTON)) {
+                Icon(
+                    imageVector = Icons.Default.ArrowBack,
```
Comment (unusedusername01) [inline app/src/main/java/com/android/sample/ui/request_validation/ValidateRequestScreen.kt]: > [!NOTE]
> Here again, arrow seems to be flagged by sonarcloud; just remind me to accept the issues if I didn't do it correctly

File: app/src/main/java/com/android/sample/ui/request_validation/ValidateRequestsConstants.kt
Comment (unusedusername01) [inline app/src/main/java/com/android/sample/ui/request_validation/ValidateRequestsConstants.kt]: Great constants and documentation !

File: app/src/androidTest/java/com/android/sample/ui/validate/ValidateRequestScreenTests.kt [hunk 5]
```diff
@@ -0,0 +1,813 @@
+package com.android.sample.ui.request_validation
```
[the remark this answers is not in the document]
Comment (Thibaud-Babin) [reply app/src/androidTest/java/com/android/sample/ui/validate/ValidateRequestScreenTests.kt]: Oops I left the mistake when I move the folder I'll just apply the mandatory 20 lashes to my back as usual (I'll also correct it while I'm at it)

File: app/src/androidTest/java/com/android/sample/ui/validate/ValidateRequestScreenTests.kt (hunk 1 above)
Comment (Thibaud-Babin) [reply app/src/androidTest/java/com/android/sample/ui/validate/ValidateRequestScreenTests.kt]: That's a good Idea I didn't think about it   I just wanted to look at a white wall for 30 hours after writing so many tests that I forgot about some edge cases , Thank you so much sensei

**Comment by Thibaud-Babin** [comment]
> Great work on this PR ! Unfortunately since it wasn't implemented in the main flow, I wasn't able to test it and to be honest I cannot really judge that much the logic since it's pretty **dense** but Ig that's work for a future PR, nice work on this one already 😎
> 
> * The utils function added to FirebaseEmulator was a clever idea although it would be nice mergine the System.currentTimeMillis() uses
> * The code of ValidateRequestScreen is pretty large and it kinda lacks comments/documentation tbh, but the I've reviewed the expected "layout" I bit and it looks sound and solid so nice work; maybe add/keep the preview screen you used next time.
> * The three dependencies are kind of "obscure", I did a research and apparently it's SQL and AR stuff but I'm not sure (pls again invite me if you're planning something big) Just remove them or document it and you'll be all good !à





 Hello favorite reviewer hihihi 🫢, I really appreciated you're review and it left me all giggly on my belly while dangling my legs. I appreciated your comments and I'll work on fixing and applying your suggestions and I'll get back to you when it's done! Thank you so much sensei !

**Review by unusedusername01 — APPROVED** [review]
Nice changes and great work on this PR !
